### PR TITLE
Initial tab completetion implementation

### PR DESCRIPTION
**Pull request by mchf**

### CODE REVIEW

**Comment by mvidner** [comment]
OK, but how do I use this?

File: tools/yast2-rake.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      COMPREPLY=( "${COMPREPLY[0]}:" )
+    else
+      COMPREPLY=( "${COMPREPLY[0]} " )
+    fi
+  fi
+
+  return 0
+}
+
+rake_cache=$(rake -T 2> /dev/null | sed "s/\[[^]]*]//") && complete -F _rake -o nospace rake
```
Comment (mvidner) [inline tools/yast2-rake.sh]: Does this mean that when I `cd` to a different Rakefile, it will use the tasks of the previous one?

Comment (mchf) [inline tools/yast2-rake.sh]: These usability details are currently being researched ;-) Thanks for hint.

**Comment by jreidinger** [comment]
general note, why it is part of yast-rake and not attach it to rake itself? I see no yast specific stuff here and I am sure other projects will also benefit from it.

**Comment by jreidinger** [comment]
mvidner - usage should be placing this sh file to /etc/bash_completion.d/, then relogin ( not sure if it is needed ) and then <tab> should work.

**Comment by mchf** [comment]
@jreidinger just restarting terminal should be enough

**Comment by lslezak** [comment]
@mchf how should we proceed here? I agree with @jreidinger as this looks generic and should be part of the rake package. (Or a separate package like `rake-bash-completion` - similar to the existing `pulseaudio-bash-completion` or `rubygem-hub-bash-completion` packages.)

On the other hand is it really needed? My usual workflow it to run `rake -T` and then copy&paste the command I need. Tab completion is nice, but when I do not know what I need to run then it does not help much.

**Comment by lslezak** [comment]
Closing, please add the completion directly to the rake package or to a separate package.

**Pull request closed without merging**